src/components/Layout/Header/Header.tsx

```diff
@@ -93,7 +95,15 @@ export const Header = memo(() => {
   )
 
   const currentWalletId = useSelector(selectWalletId)
+  const walletAccountIds = useSelector(selectWalletAccountIds)
+
   useEffect(() => {
+    if (previousSnapInstall === null && isSnapInstalled === false) {
+      // We have just detected that the user doesn't have the snap installed currently
+      // We need to check whether or not the user had previous non-EVM AccountIds and clear those
+      if (currentWalletId && walletAccountIds.some(accountId => isUtxoAccountId(accountId)))
```

Comment:
seems like we only want to do this if the current wallet is metamask, not any wallet with utxo accounts, unless I am misunderstanding?

Reply:
Oh yeah, absolutely. We should never run this whole effect if the current wallet isn't MetaMask, and `MetaMaskShapeShiftMultiChainHDWallet` specifically.

Reply:
29e0d7523eeab2fe9065a5bbd390de7256cc5a37